### PR TITLE
industry model added, views and urls implemented & profile updated to…

### PR DESCRIPTION
**Pull request by blakerach1**
## Describe your changes

Inserted Industry & Tag model
Industry & Tag serializer added
Industry views implemented - GET (list), POST (only admin user) & DELETE (only admin user)
Tag views implemented - GET (list), POST (authenticated user only) - unique and similar tags not allowed, & DELETE (admin only)

Urls added:
- industries/ - GET (list) and POST available through this
- industries/<int:pk> - DELETE available through this
- tags/ - GET (list) - no permissions and POST (authorised user only)
- tag/<int:pk> - DELETE available (admin user only)
Updated the profile serializer & model to include the industries & tag data

Checks performed:
list of industry category returned
admin user only can create industry category
admin user only can delete industry category
unauthorised user cannot create or delete industry category

list of tags returned
authenticated user only can create tag - tag cannot be the same or similar as one already in the database - data is not transformed to save to database, but rather the incoming data is transformed and checked against similar results in database
admin user only can delete tag

**Improvements still to be made:**

Need to inform checks for similar titles on industries - i.e. DevOps and Dev Ops. At the moment only the unique validator is applied. Should be able to achieve now that i've implemented on tags. 

Need to limit categories selected on a profile to no more than three. I have asked query in Slack. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## What kind of feedback do you need?

Review of Industry model set up - including models.py, urls.py, serializers.py and views.py.
Additionally a review of the incorporation of Industry into the existing profile model. 

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have recommended where to start and how to proceed with the review

PUT industry added to profile
GET industry added to profile
DELETE industrie_admin_approved
DELETE industries_permission_req
POST industries_admin_approved
POST industries_admin_permission_req
GET industries_no_permission

Tag checks in Insomnia:
GET profile incorporated check
DELETE tag
POST tag similarity check 2
POST tag similarity check
POST tags
GET tags

### CODE REVIEW

**Review by shayzimm — APPROVED** [review]
Amazing work, Rachel!

File: DiversiTech/profiles/views.py
Comment (Lyndallh) [inline DiversiTech/profiles/views.py:1]: I'm wondering about the cascade user delete on profile delete. Would this be the where you'd put it? Or does it need to come from the front end?

**Review by Lyndallh — APPROVED** [review]

**Comment by Lyndallh** [comment]
This is amazing work, such a big piece of work. Well done :) :)

File: DiversiTech/profiles/views.py [hunk 1]
```diff
@@ -63,5 +147,5 @@ def put (self, request, pk):
     def delete(self,request, pk):
         profile = self.get_object(pk)
         profile.delete()
-        return Response({"message":"Profile deleted successfully"}, status=status.HTTP_204_NO_CONTENT)
+        return Response({"message":"Profile deleted successfully"}, status=status.HTTP_200_OK)
```
Comment (Lyndallh) [inline DiversiTech/profiles/views.py:151]: I'm wondering if this is where you'd include the user delete on a cascade of the profile delete? or would it come from some front end logic?

**Review by Lyndallh — APPROVED** [review]

File: DiversiTech/profiles/views.py (hunk 1 above)
Comment (blakerach1) [reply DiversiTech/profiles/views.py:151]: Hey Lindi, we've got on_delete=cascade in the model for profile, next to the foreign key for user. I believe that means any profile delete will cause user to be deleted and vice versa

**Pull request merged**